core_java/grep/ca/jrvs/apps/grep/JavaGrepImp.java

```diff
@@ -0,0 +1,124 @@
+import java.util.*;
+import java.io.*;
+import java.lang.*;
+import java.util.regex.Pattern;
+
+import com.sun.org.slf4j.internal.Logger;
+import com.sun.org.slf4j.internal.LoggerFactory;
+import org.apache.log4j.BasicConfigurator;
+
+public class JavaGrepImp implements JavaGrepp {
+
+  final Logger logger = LoggerFactory.getLogger(JavaGrepp.class);
+
+  private String regex;
+  private String rootPath;
+  private String outFile;
+
+  public static void main(String[] args) {
+    if (args.length != 3) {
+      throw new IllegalArgumentException("Usage: JavaGrep regex rootPath outFile");
+    }
+
+    // Use default logger config
+    BasicConfigurator.configure();
+
+    JavaGrepImp javaGrepImp = new JavaGrepImp();
+    javaGrepImp.setRegex(args[0]);
+    javaGrepImp.setRootPath(args[1]);
+    javaGrepImp.setOutFile(args[2]);
+
+    try {
+      javaGrepImp.process();
+    } catch (Exception ex) {
+      javaGrepImp.logger.error(ex.getMessage(), ex);
```

Comment:
javaGrepImp.logger.error("meaningfull app specific messages, ex);

you don't want to double log the same message